lib/krb5/config_file.c

```diff
@@ -553,8 +563,7 @@ krb5_config_parse_file_multi (krb5_context context,
 	if (ret) {
 	    krb5_set_error_message(context, ret,
 				   "Failed to parse plist %s", fname);
-	    if (newfname)
-		free(newfname);
+            free(newfname);
 	    return ret;
 	}
 #else
```

Comment:
The `#else` codepath is, I think, missing a decrement of `context->config_include_depth`